examples/HowTo.md

```diff
@@ -0,0 +1,147 @@
+
+```haskell
+mainWithLogging :: IO ()
+mainWithLogging = do
+    buildAndSetupYamlLogging productionB "examples/how-to-log-config.yaml"
+    usingLoggerName "new-logger" inputLengthWithLog
+```
+where we set up the config from the file and run `inputLengthWithLog` under the logger with name `new-logger`.
+
+Let's have a closer look at how we transform the `inputLength` function to the `inputLengthWithLog` that would work with logging in the next paragraph.
+
+## Add logging messages to function in `IO`
+
+If you want to use logging in a function that already runs in the `IO` monad, replace the monad with [`WithLoggerIO`](https://hackage.haskell.org/package/log-warper-1.7.1/docs/System-Wlog-CanLog.html#t:WithLoggerIO) and `liftIO` all your `IO` actions.
+
+For example, our `inputLength` function works this way:
+```haskell
+
+inputLength :: IO ()
+inputLength = do
+    input <- TIO.getLine
+    case T.length input of
+        0 -> putStrLn "Should not be empty" >> inputLength
+          -- v ^  you need to log these
+        _ -> putStrLn "Well done" >> funWithInput input
+
+```
+Very easy, nothing interesting. We have some input from console and write some messages depending of the length of user's input.
+What we are going to do is to log this messages instead of just `putStrLn` them.
+
+First that we should consider is that logging can not be done in `IO`, so we need make the function work in `WithLoggerIO` monad,
+so that we are able to do any logging stuff inside.
+After this is done you can easily use `logDebug`, `logInfo`, `logWarning` etc.
+(see others [here](https://hackage.haskell.org/package/log-warper-1.7.1/docs/System-Wlog-CanLog.html#v:logDebug)) to log messages.
+As it's not `IO` after our transformations, we shouldn't forget to `liftIO` functions that work in `IO` and doesn't have logging inside.
+As result we will get
+
+```haskell
+
+inputLengthWithLog :: WithLoggerIO m => m ()
+inputLengthWithLog = do
+    input <- liftIO TIO.getLine
+    case T.length input of
+        0 -> do
+            logError "Should not be empty"
+            inputLengthWithLog
+        _ -> do
+            logInfo "Well done"
+            funWithInputInMonad input
+
+```
+
+Another moment, let's have a closer look at the `funWithInput` function from the example above.
+We see that its type is `IO ()` and it doesn't have logging inside.
+
+```haskell
+funWithInput :: Text -> IO ()
+funWithInput input = do
+    TIO.putStrLn $ "Wow! You wrote " <> input
+
+```
+So we also can rewrite it in this way
+```haskell
+funWithInputInMonad :: MonadIO m => Text -> m ()
+funWithInputInMonad input =
+    liftIO $ TIO.putStrLn $ "Wow! You wrote " <> input
+
+```
+or usage of `liftIO` is also the solution.
+
+And as a conclusion, let's write the `main` function to check how we managed to apply logging in very simple example
+```haskell
+main :: IO ()
+main = do
+    mainWithoutLogging
+    mainWithLogging
+```
```

Comment:
I have very cool idea: at the end of tutorial you can add two-cell table (like in this project: https://github.com/serokell/importify/blob/master/README.md) with IO on the left and `WithLoggerIO` on the right. The only difficulty is whether it will affect compilation of this file... Maybe `markdown-unlit` has some special pragma with which you can disable compilation of some part of Markdown, or maybe it doesn't compile tables automatically, or maybe you will need to put this table into separate file...

Reply:
Yes, there is `ignore` mark specially for these cases when you don't won't something to be compiled, so it would be cool to have the table! :heart_eyes:

Reply:
Well, the thing is that it's not possible to put code blocks into markdown tables (or I missing something..) :disappointed:

Reply:
Yes, this is really a problem, indeed... I think you can use inline code blocks and find a way to use space characters there. I think that'll do for now.